api/post.go

```diff
@@ -705,7 +705,7 @@ func sendNotifications(c *Context, post *model.Post, team *model.Team, channel *
 			}
 
 			if userAllowsEmails && status.Status != model.STATUS_ONLINE {
-				go sendNotificationEmail(c, post, profileMap[id], channel, team, senderName[id], sender)
+				sendNotificationEmail(c, post, profileMap[id], channel, team, senderName[id], sender)
```

Comment:
Ya I have a separate ticket to look into this. I think what we are going to do here is a fixed length queue that blocks if full (buffered go channel)